src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -225,17 +226,20 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     IEnumerable<string> structuralProperties = structuredType.StructuralProperties()
                         .Select(edmProperty => EdmLibHelpers.GetClrPropertyName(edmProperty, model));
 
+                    IEnumerable<string> updatableProperties= structuralProperties.Union(structuredType.NavigationProperties()
+                        .Select(edmProperty => EdmLibHelpers.GetClrPropertyName(edmProperty, model)));
```

Comment:
The StructuralProperties() extension method gets the StructuredDefinition for the structuredType and then loops through all properties that are OfType<IEdmStructuralProperty>.  NavigationProperties() does the same for navigation properties.  So the end result of these four lines is to filter the properties first by structural property, then by navigation property, and then union them back together. 

Instead, simply Replace 226-230 with:
IEnumerable<string> structuralProperties = structuredType.StructuredDefinition().Properties()
                        .Select(edmProperty => EdmLibHelpers.GetClrPropertyName(edmProperty, model));